deps/buffer.lua

```diff
@@ -49,11 +49,11 @@ local C = ffi.os == "Windows" and ffi.load("msvcrt") or ffi.C
 function Buffer:initialize(length)
   if type(length) == "number" then
     self.length = length
-    self.ctype = ffi.gc(ffi.cast("unsigned char*", C.malloc(length)), C.free)
+    self.ctype = ffi.gc(ffi.cast("unsigned char*", C.calloc(1, length)), C.free)
```

Comment:
Just a correctness thing, the size should be `1` and the length should be `length`. You're allocating `length` 1-byte items. Not 1 `length`-byte items

Reply:
Hm, not sure if that analogy makes more sense to me than the other, mind explaining more on why is it "Allocating `length` 1-byte items` instead of it being a single `length` item?

Reply:
Because it's an array of bytes (`size=1`), not a single `size=len` item.